chirp/drivers/retevis_ra25.py

```diff
@@ -0,0 +1,1098 @@
+    char    id[16];
+} remote_stun;
+
+struct{
+    char    id[16];
+} remote_kill;
+
+struct{
+    u8      scanmode;
+    u8      prio_chan;
+    u8      revert_chan;
+    u8      lookback_a;
+    u8      lookback_b;
+    u8      dropout_delay;
+    u8      dwell;
+    u8      unk_sm;
+    u8      unused[8];
+} scan_menu;
+
+struct{
+    char    message[16];
+}startup_display;
+
+struct{
+    u8      unk_imm[16];
+    char    serial_number[16];
+} immut_1;
+
+struct{
+    char    prod_date[10];
+    u8      unk_imm2[22];
+} immut_2;
+"""
+
+SUPPORTED_MODEL_VERSIONS = {"RA_25UV": ["V200"]}
+
+# RADIO MODES
+BAND_LABELS = [
+    "UHF ( 400 - 490 MHz ) + VHF( 136 - 174 MHz )",
+    "UHF ( 400 - 490 MHz ) + VHF( 144 - 146 MHz )",
+    "UHF ( 430 - 440 MHz ) + VHF( 136 - 174 MHz )",
+    "UHF ( 430 - 440 MHz ) + VHF( 144 - 146 MHz )",
+    "UHF ( 430 - 440 MHz ) + VHF( 144 - 148 MHz )",
+    "UHF ( 400 - 438 MHz ) + VHF( 136 - 174 MHz )",
+    "UHF ( 420 - 450 MHz ) + VHF( 144 - 148 MHz )",
+    "UHF ( 400 - 470 MHz ) + VHF( 136 - 174 MHz )",
+    "European Version PMR",
+    "US GMRS",
+    "Australian UHF CB"
+]
+BAND_LIMITS = [
+    [(136000000, 174000001), (400000000, 490000001)],
+    [(144000000, 146000001), (400000000, 490000001)],
+    [(136000000, 174000001), (430000000, 440000001)],
+    [(144000000, 146000001), (430000000, 440000001)],
+    [(144000000, 148000001), (430000000, 440000001)],
+    [(136000000, 174000001), (400000000, 438000001)],
+    [(144000000, 148000001), (420000000, 450000001)],
+    [(136000000, 174000001), (400000000, 470000001)],
+    [(446000000, 446200001)],
+    [(462550000, 467725001)],
+    [(476425000, 477412501)]
+]
+
+MEMORY_REGIONS = [
+    ('00022000', 2, 32, 'W', 'chans_present', 'Present Chan.'),
+    ('00032000', 2, 32, 'W', 'chans_scanned', 'Scanned Chan.'),
+    ('00042000', 200, 32, 'W', 'memory', 'Memory Chan. 1-200'),
+    ('00052000', 200, 32, 'S', 'memory2', 'Memory Chan. 201-400'),
+    ('00062000', 100, 32, 'S', 'memory3', 'Memory Chan. 401-500'),
+    ('01002000', 1, 32, 'W', 'func_settings', 'Function Menu'),
+    ('02001000', 16, 16, 'W', 'dtmf_list', 'DTMF Encode List'),
+    ('02011000', 1, 16, 'W', 'dtmf_menu', 'DTMF Menu'),
+    ('02031000', 1, 16, 'W', 'dtmf_start', 'DTMF PTT ID Start'),
+    ('02041000', 1, 16, 'W', 'dtmf_end', 'DTMF PTT ID End'),
+    ('02071000', 1, 16, 'W', 'remote_stun', 'DTMF Remote Stun'),
+    ('02081000', 1, 16, 'W', 'remote_kill', 'DTMF Remote Kill'),
+    ('03011000', 1, 16, 'W', 'scan_menu', 'Scan Menu'),
+    ('03021000', 1, 16, 'W', 'startup_display', 'Startup Display'),
+    ('08002000', 1, 32, 'RO', 'immut_1', 'Serial Number'),
+    ('08002001', 1, 32, 'RO', 'immut_2', 'Production Date')
+]
+
+
+def send_cmd(serial, cmd):
+    try:
+        resp = b''
+        serial.write(cmd)
+        serial.flush()
+        timeout = time() + SERIAL_TIMEOUT
+        while time() < timeout:
+            if serial.inWaiting():
+                resp += serial.read()
```

Comment:
Please don't do this. The serial layer already lets you read the amount of data you want with a timeout. Code that tries to read non-constant responses is almost always very fragile on different platforms. Please set a longish timeout (like 1s) and read exactly the number of bytes you expect for each response.